bindings/azure/eventgrid/eventgrid.go

```
@@ -0,0 +1,191 @@
+type AzureEventGrid struct {
+	metadata *azureEventGridMetadata
+	logger   logger.Logger
+}
+
+type azureEventGridMetadata struct {
+	TenantID                  string `json:"tenantId"`
+	SubscriptionID            string `json:"subscriptionId"`
+	ResourceGroupName         string `json:"resourceGroupName"`
+	TopicName                 string `json:"topicName"`
+	EventGridSubscriptionName string `json:"eventGridSubscriptionName"`
+	SubscriberEndpoint        string `json:"subscriberEndpoint"`
+	TopicEndpoint             string `json:"topicEndpoint"`
+	ClientID                  string `json:"clientId"`
+	ClientSecret              string `json:"clientSecret"`
+	HandshakePort             string `json:"handshakePort"`
+}
+
+// NewAzureEventGrid returns a new Azure Event Grid instance
+func NewAzureEventGrid(logger logger.Logger) *AzureEventGrid {
+	logger.Debug("NewAzureEventGrid() called...")
+	return &AzureEventGrid{logger: logger}
+}
+
+// Init performs metadata init
+func (a *AzureEventGrid) Init(metadata bindings.Metadata) error {
+	a.logger.Debugf("Parsing Event Grid metadata(%s)...", metadata.Name)
+
+	m, err := a.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	a.metadata = m
+
+	a.logger.Debug("Metadata parsed successfully.")
+
+	return nil
+}
+
+func (a *AzureEventGrid) Read(handler func(*bindings.ReadResponse) error) error {
+	a.logger.Debug("Read() called...")
+
+	err := a.createSubscription()
+	if err != nil {
+		return err
+	}
+
+	m := func(ctx *fasthttp.RequestCtx) {
+		switch string(ctx.Path()) {
+		case "/api/events":
+			if string(ctx.Method()) == "OPTIONS" {
+				ctx.Response.Header.Add("WebHook-Allowed-Origin", string(ctx.Request.Header.Peek("WebHook-Request-Origin")))
+				ctx.Response.Header.Add("WebHook-Allowed-Rate", "*")
+				ctx.Response.Header.SetStatusCode(fasthttp.StatusOK)
+				_, err = ctx.Response.BodyWriter().Write([]byte(""))
+				if err != nil {
+					a.logger.Error(err.Error())
```

Does it mean we are swallowing errors?

---

Unfortunately, I don't believe there is anything we can do here besides log the error.